### ISSUE TITLE
Sha1 / base64

### ISSUE THREAD

**Issue by lequarti** [theturtle32/AS3WebSocket#1]
Hi, 

For my purpose, I used Gsolo / Crypto / Sha 1 (http://gsolofp.blogspot.com/2006/01/actionscript-3-md5-and-sha1.html) 
for the handshake, cuz my phpServer, send sha1 in raw format (binary 20 lenght)

**Comment by theturtle32** [comment]
I'm not sure why you opened an issue for this?  Using a binary SHA-1 hash in the WebSocket handshake is wrong and violates the specification.  There is no reason why your WebSocket server cannot use a Base64 encoded hash, and indeed should do so.  You should not modify a WebSocket client in the way you describe.

**Comment by lequarti** [comment]
Sorry It's not an issue, I agree, I didn't found another way so I've put it here.

Well my application driven by a php socket server have to be compatible withe chrome (14), firefox (7) and Adobe Air (3) with your Lib. 

The server interact well with the browser but not with lib, so by gsolo/sha1 it worked fine with no problem. In fact I pickgsolo based on websocket-js the flash bridge.

sorry I troubled here.

Best regards.

**Comment by theturtle32** [comment]
No trouble at all.  I'm curious about the problem you had.. can you give me more information?  What php server are you using?  I wrote the AS3 client to be compatible with the spec, and it seems to interoperate just fine with the other open source WebSocket implementations that I've tried.

In fact, if a browser accepts a raw binary SHA-1 hash in the Sec-WebSocket-Accept header, I would consider that to be a bug in the browser.  Are you certain that it's being transmitted as raw binary?

**Comment by lequarti** [comment]
> Are you certain that it's being transmitted as raw binary?

May be I'm using the wrong term.

Well here the informations about my app framework :

I use github / GulDmitry / php-websocket-server

Here the portion of the code to generation the accept : 

return base64_encode( sha1( $headers['Sec-WebSocket-Key'] . '258EAFA5-E914-47DA-95CA-C5AB0DC85B11', true) );

It works fine with native websocket under Chrome 14 and Firefox 7; but not with AS3Ws.... lib !

According the PHP Doc : http://fr.php.net/manual/en/function.sha1.php, I set to false the second argument of SHA1 function, it works with AS3Ws.... but not with Chrome & Firefox, was a bit confused.

Then I read into the code of github / gimite / web-socket-js, I saw the using of Gsolo Sha1 class instead of As3corelib, I didtn't want to alterate your code too much, I juste import those classes and added this 3 lines :

```
    //import com.adobe.crypto.SHA1;
    import com.gsolo.encryption.SHA1;
```

```
                        SHA1.b64pad = "=";
                        var handleHash:String  = base64nonce.concat("258EAFA5-E914-47DA-95CA-C5AB0DC85B11");
                        var expectedKey:String = SHA1.b64_sha1(handleHash);
```

And all was fine, my app can interacte both with browser and air app.

Voilà.
